src/evict/evict_page.c

```diff
@@ -663,34 +663,64 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     /* Make sure that both conditions above are not true at the same time. */
     WT_ASSERT(session, !use_snapshot_for_app_thread || !is_eviction_thread);
 
-    if (!conn->txn_global.checkpoint_running && !WT_IS_HS(btree->dhandle) &&
-      (use_snapshot_for_app_thread || is_eviction_thread)) {
+    /* History store data is always evictable. */
```

Comment:
Can you add a warning comment here that changing these if cases could result in a unforeseen consequences or similar? We need to make it clear that this code you shouldn't change without a very well thought out reason.

Reply:
Something like "Take care when modifying the following logic as there are a number of edge cases here that aren't immediately obvious."

Reply:
Do we have such kind of precedence? I think that just discourages people from making the changes.

Reply:
We do have similar comments in other places that are along the same lines, and yes the intention is to discourage people from making changes without a good idea of said edge case are.

Reply:
The intention of the refactor is to make such cases obvious? Do you think if there is still any subtle edge cases following the new code flow?

Reply:
We are modifying this code to handle a race that can occur between eviction and checkpoint which results in the history store having records removed incorrectly. That case isn't mentioned in any of the following comments explicitly.

Reply:
added